sentry/src/routes/campaign.rs

```diff
@@ -217,6 +218,38 @@ pub async fn create_campaign<A: Adapter>(
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
 
+pub async fn campaign_list<A: Adapter>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let mut query =
+        serde_urlencoded::from_str::<CampaignListQuery>(req.uri().query().unwrap_or(""))?;
+
+    query.validator = match (query.validator, req.extensions().get::<Auth>()) {
+        (Some(validator), _) => Some(validator),
+        (None, Some(session)) => Some(session.uid),
+        (None, None) => None,
+    };
```

Comment:
`Auth` should be used as `query.validator` only when `is_leader` is passed.
By default it should not filter results by validator.